### PR TITLE
Fix: Treat empty base constructor argument list as not provided.

### PR DESCRIPTION
**Pull request by chriseth**
Fixes https://github.com/ethereum/solidity/issues/2912

### CODE REVIEW

File: test/libsolidity/syntaxTests/inheritance/base_arguments_empty_parentheses.sol [hunk 1]
```diff
@@ -0,0 +1,5 @@
+contract Base {
+  function Base(uint) public {}
+}
+contract Derived is Base(2) { }
+contract Derived2 is Base(), Derived() { }
```
Comment (axic) [inline test/libsolidity/syntaxTests/inheritance/base_arguments_empty_parentheses.sol:5]: Can you add `contract Derived3 is Base, Derived {}` to make sure the syntactical difference wont cause any issue down the line?

File: libsolidity/codegen/ContractCompiler.cpp [hunk 2]
```diff
@@ -238,6 +239,7 @@ void ContractCompiler::appendBaseConstructor(FunctionDefinition const& _construc
 		solAssert(m_baseArguments.count(&_constructor), "");
 		std::vector<ASTPointer<Expression>> const* arguments = m_baseArguments[&_constructor];
 		solAssert(arguments, "");
+		solAssert(arguments->size() == constructorType.parameterTypes().size(), "");
```
Comment (axic) [inline libsolidity/codegen/ContractCompiler.cpp:242]: Do we have a test case for this? I think it may silently accept less parameters at the moment.

**Review by axic — CHANGES_REQUESTED** [review]

File: libsolidity/codegen/ContractCompiler.cpp (hunk 2 above)
Comment (axic) [reply libsolidity/codegen/ContractCompiler.cpp:242]: It fails properly, but we actually don't have a test for it.

Comment (chriseth) [reply libsolidity/codegen/ContractCompiler.cpp:242]: Added.

**Review by axic — APPROVED** [review]
We can squash this.

**Pull request merged**